rubocop-shopify.gemspec

```diff
@@ -21,5 +21,5 @@ Gem::Specification.new do |s|
     "allowed_push_host" => "https://rubygems.org",
   }
 
-  s.add_dependency("rubocop", ">= 0.81.0")
+  s.add_dependency("rubocop", "~> 0.82.0")
```

Comment:
cc @rafaelfranca I remember you said we would be lax in here to be aware about breaking changes in upcoming RuboCop versions. But since now we have Dependabot we can be more strict about the version of RuboCop we support to avoid breaking users' setups. Wdyt?

Reply:
I still think upper limits in any gem is a waste of time. It just annoys users. If users want to use a new version of rubocop and we for some reason are too busy to merge the PR on this gem and release a new version they will not be able to do.

What do you expect to get from the upper limit? What exactly are we protecting users of?

Reply:
> What do you expect to get from the upper limit? 

I'm trying to be as clear as possible about the compatibility of this gem's RuboCop config. I'm trying to say "the current version works with RuboCop 0.82.0 and its additional patches, but we can't guarantee it's gonna work for 0.83"

With Dependabot we will bump this as soon as a new RuboCop version is out, in which we can test and release accordingly.

> What exactly are we protecting users of?

I thing it's a good practice to let users run a global `bundle update` that won't break their RuboCop setup. If we are overly permissive with the versioning they could upgrade to a newer RuboCop version that we are not yet compatible with.

Again, because Dependabot, we will get an immediate PR to bump and fix incompatibilities, so it's not like we are going to prevent people from upgrading to the latest RuboCop or anything. Worst case would be if we end up taking too long to merge Dependabot's PRs, but that scenario would also be possible for PRs opened by users.